### PR TITLE
Transfer EnvironmentService to Jellyfin.Api

### PR DESCRIPTION
**Pull request by crobibero**
According to the following `ValidatePath` only uses a post body.
https://github.com/jellyfin/jellyfin-web/blob/ef7cf6c24bf67305e3ecca69739b1e2651165191/src/components/directorybrowser/directorybrowser.js#L156

Part of #2872

### CODE REVIEW

File: Jellyfin.Api/Models/EnvironmentDtos/DefaultDirectoryBrowserInfo.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+namespace Jellyfin.Api.Models.EnvironmentDtos
+{
+    /// <summary>
+    /// Default directory browser info.
+    /// </summary>
+    public class DefaultDirectoryBrowserInfo
```
Comment (cvium) [inline Jellyfin.Api/Models/EnvironmentDtos/DefaultDirectoryBrowserInfo.cs:6]: Lacking the Dto suffix.

File: Jellyfin.Api/Controllers/EnvironmentController.cs [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+                }
+            }
+            else
+            {
+                if (!System.IO.File.Exists(validatePathDto.Path) && !Directory.Exists(validatePathDto.Path))
+                {
+                    return NotFound();
+                }
+
+                if (validatePathDto.ValidateWritable)
```
Comment (JustAMan) [inline Jellyfin.Api/Controllers/EnvironmentController.cs:115]: wtf is this doing, why is it only done when `IsFile` is null, and why is it done if `dto.Path` points to a file?

File: Jellyfin.Api/Controllers/EnvironmentController.cs [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+        /// Gets network paths.
+        /// </summary>
+        /// <response code="200">Empty array returned.</response>
+        /// <returns>List of entries.</returns>
+        [Obsolete("This endpoint is obsolete.")]
+        [HttpGet("NetworkShares")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public ActionResult<IEnumerable<FileSystemEntryInfo>> GetNetworkShares()
+        {
+            return Array.Empty<FileSystemEntryInfo>();
```
Comment (JustAMan) [inline Jellyfin.Api/Controllers/EnvironmentController.cs:145]: can we log a warning here for accessing an obsolete endpoint?

File: Jellyfin.Api/Controllers/EnvironmentController.cs [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+        /// <returns>Parent path.</returns>
+        [HttpGet("ParentPath")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public ActionResult<string?> GetParentPath([FromQuery, BindRequired] string path)
+        {
+            string? parent = Path.GetDirectoryName(path);
+            if (string.IsNullOrEmpty(parent))
+            {
+                // Check if unc share
+                var index = path.LastIndexOf(UncSeparator);
```
Comment (JustAMan) [inline Jellyfin.Api/Controllers/EnvironmentController.cs:178]: this feels wrong... UNC paths should _start_ with double slash, not start with a slash and have a slash _somewhere_

File: Jellyfin.Api/Controllers/EnvironmentController.cs [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+        /// <param name="includeFiles">An optional filter to include or exclude files from the results. true/false.</param>
+        /// <param name="includeDirectories">An optional filter to include or exclude folders from the results. true/false.</param>
+        /// <response code="200">Directory contents returned.</response>
+        /// <returns>Directory contents.</returns>
+        [HttpGet("DirectoryContents")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public IEnumerable<FileSystemEntryInfo> GetDirectoryContents(
+            [FromQuery, BindRequired] string path,
+            [FromQuery] bool includeFiles,
+            [FromQuery] bool includeDirectories)
```
Comment (JustAMan) [inline Jellyfin.Api/Controllers/EnvironmentController.cs:50]: in original API those were optional, now they seem to be not... looks wrong

File: Jellyfin.Api/Controllers/EnvironmentController.cs [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+        [HttpPost("ValidatePath")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public ActionResult ValidatePath([FromBody, BindRequired] ValidatePathDto validatePathDto)
+        {
+            if (validatePathDto.IsFile.HasValue)
+            {
+                if (validatePathDto.IsFile.Value)
+                {
+                    if (!System.IO.File.Exists(validatePathDto.Path))
```
[the remark this answers is not in the document]
Comment (crobibero) [reply Jellyfin.Api/Controllers/EnvironmentController.cs:95]: This was direct copy from EnvironmentService. I would switch to _fileSystem but there is no `Exists` function

File: Jellyfin.Api/Controllers/EnvironmentController.cs (hunk 2 above)
Comment (crobibero) [reply Jellyfin.Api/Controllers/EnvironmentController.cs:115]: I think the goal was to determine if a path was writable. Once we start retooling the api functionality we can make possibly breaking changes

File: Jellyfin.Api/Controllers/EnvironmentController.cs (hunk 4 above)
Comment (crobibero) [reply Jellyfin.Api/Controllers/EnvironmentController.cs:178]: This might be testing for nested UNC paths?

**Pull request merged**